File: 04-estructuras-datos/13-seleccion-y-combinacion-estructuras/java/ejemplos/CombinarEstructuras.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

public class CombinarEstructuras {
    record Curso(String codigo, String nombre) {}

    public static void main(String[] args) {
        List<Curso> ordenRegistro = new ArrayList<>();
        Map<String, Curso> porCodigo = new HashMap<>();
        Set<String> categorias = new HashSet<>();

        Curso curso = new Curso("JAVA-01", "Java básico");
        ordenRegistro.add(curso);       // List conserva el orden de registro.
        porCodigo.put(curso.codigo(), curso); // Map permite localizar por clave.
        categorias.add("Programación"); // Set evita categorías duplicadas.

        System.out.println(porCodigo.get("JAVA-01"));
    }
}

File: 04-estructuras-datos/13-seleccion-y-combinacion-estructuras/scala/ejemplos/CombinarEstructuras.scala
case class Curso(codigo: String, nombre: String)

object CombinarEstructuras:
  def main(args: Array[String]): Unit =
    val cursos = Vector(Curso("SCALA-01", "Scala básico"))
    val porCodigo = cursos.map(c => c.codigo -> c).toMap
    val categorias = Set("Programación")
    // Cada estructura conserva una responsabilidad diferente.
    println(porCodigo("SCALA-01"))
    println(categorias)
